### DISCUSSION TITLE
Rate limit "from.lower" for outgoing mails only

### DISCUSSION THREAD

**Discussion by leofeyer** [rspamd/rspamd#3758] [Q&A]
I am trying to set up a rate limit by sender for **all** outgoing mails. The `user` rule covers authenticated users delivering their mails via SMTP, however, I also want to handle local mails delivered via sendmail. Therefore, I have added a rule using `from.lower` as selector:

```
# local.d/ratelimit.conf
rates {
  sender_limit = {
    selector = 'user.lower';
    bucket = {
      burst = 100;
      rate = 0.01666666666666666666; # leak 1 message per minute
    }
  }
}
```

Unfortunately, this limit is also applied to **incoming** mails and thus adding a lot of unnecessary entries to the redis storage. Is there a way to limit the rule to outgoing mails only?

Thank you for your help.

**Comment by fatalbanana** [comment]
You could use `user.id('');from.lower` as selector.

https://rspamd.com/doc/configuration/selectors.html

**Comment by leofeyer** [comment]
Thank you for the hint, however, I do not understand how this would solve the problem. `user.id('')` prevents applying the rule to non-authenticated users, right? But outgoing mails delivered via sendmail do not have an authenticated user and I still want to rate limit them by sender.

**Reply by fatalbanana** [reply]
So how might you identify the mail as outbound?

For more complex use-cases writing one's own selectors in Lua is possible.

**Reply by leofeyer** [reply]
> So how might you identify the mail as outbound?

That is exactly my problem. Postfix is really missing a `non_smtpd_sender_restrictions` configuration.
